Containerized Ubiquity-Docker-Plugin

aswarke opened the pull request:
Containerize ubiquity-docker-plugin as per Docker's managed plugin system ( plugin v2 architecture)

Pull request supporting
1) Issue #52 
2) Issue #51 
3) IBM/ubiquity#126
4) IBM/ubiquity#127 (awaiting SCBE changes if any)
5) IBM/ubiquity#125

TODO: Will change glide.yaml ubiquity dependency to dev branch once capabilities branch has been checked into ubiquity/dev branch

shay-berman commented:
Review status: 0 of 12 files reviewed at latest revision, 19 unresolved discussions.

---

*[config.json, line 3 at r3](https://reviewable.io:443/reviews/ibm/ubiquity-docker-plugin/53#-Kw5t-MAls8kxMkLwaxd:-Kw5t-MAls8kxMkLwaxe:b-cazqsg) ([raw file](https://github.com/ibm/ubiquity-docker-plugin/blob/758ce358f0e128e4edacb736f0feb495eee39775/config.json#L3)):*
> ```JSON
> {
>   "description": "Ubiquity Docker Volume Plugin",
>   "documentation": "https://github.com/IBM/ubiquity-docker-plugin",
> ```

This is problematic
The documentation for GA is going to be in SCBE user guide, so the right location here should be change according for the GA

---

*[config.json, line 76 at r3](https://reviewable.io:443/reviews/ibm/ubiquity-docker-plugin/53#-Kw5tYvbvuIxuejqsr8e:-Kw5tYvbvuIxuejqsr8f:b-jvq5gq) ([raw file](https://github.com/ibm/ubiquity-docker-plugin/blob/758ce358f0e128e4edacb736f0feb495eee39775/config.json#L76)):*
> ```JSON
>     "socket": "ubiquity.sock",
>     "types": [
>       "docker.volumedriver/1.0"
> ```

r u sure about this version 1.0?

---

*[config.json, line 95 at r3](https://reviewable.io:443/reviews/ibm/ubiquity-docker-plugin/53#-Kw5tdDo6vzp0JfNTTzz:-Kw5tdDo6vzp0JfNTU--:b-fl1pbo) ([raw file](https://github.com/ibm/ubiquity-docker-plugin/blob/758ce358f0e128e4edacb736f0feb495eee39775/config.json#L95)):*
> ```JSON
>     },
>     {
>       "source": "/root/.ssh",
> ```

what is this for? 
if its for Spectrum Scale please add docstring here to mention it

---

*[config.json, line 113 at r3](https://reviewable.io:443/reviews/ibm/ubiquity-docker-plugin/53#-Kw5tkJY_C5i9opXzdIU:-Kw5tkJY_C5i9opXzdIV:bfwf3y4) ([raw file](https://github.com/ibm/ubiquity-docker-plugin/blob/758ce358f0e128e4edacb736f0feb495eee39775/config.json#L113)):*
> ```JSON
>     },
>     {
>       "source": "/etc/passwd",
> ```

r u sure we need this bind?  why?  the contain run as root so why we need it?

---

*[Dockerfile, line 10 at r3](https://reviewable.io:443/reviews/ibm/ubiquity-docker-plugin/53#-Kw5tvQzTVJ3BFID4Ni7:-Kw5tvQzTVJ3BFID4Ni8:byv12dn) ([raw file](https://github.com/ibm/ubiquity-docker-plugin/blob/758ce358f0e128e4edacb736f0feb495eee39775/Dockerfile#L10)):*
> ```Dockerfile
> 
> FROM alpine:latest
> RUN apk --update add ca-certificates multipath-tools nfs-utils open-iscsi openssh sg3_utils \
> ```

you should mention package=VERSION, so it will take a specific version and not the latest.
so please checkout what are the latest one that it install it with, and just put the vesion inside.  (we want to have full control on the version that comes to GA version)

---

*[glide.yaml, line 6 at r3](https://reviewable.io:443/reviews/ibm/ubiquity-docker-plugin/53#-Kw5uOBr9gt-iyGeXwZF:-Kw5uOBr9gt-iyGeXwZG:b-k5p2sq) ([raw file](https://github.com/ibm/ubiquity-docker-plugin/blob/758ce358f0e128e4edacb736f0feb495eee39775/glide.yaml#L6)):*
> ```YAML
>   version: ^1.3.0
> - package: github.com/IBM/ubiquity
>   version: capabilities
> ```

change it to dev before the marj

---

*[main_suite_test.go, line 80 at r3](https://reviewable.io:443/reviews/ibm/ubiquity-docker-plugin/53#-Kw5ulZ6mG0kEAFvqAa_:-Kw5ulZ6mG0kEAFvqAaa:b-x7euif) ([raw file](https://github.com/ibm/ubiquity-docker-plugin/blob/758ce358f0e128e4edacb736f0feb495eee39775/main_suite_test.go#L80)):*
> ```Go
> 	backend = "spectrum-scale"
> 
> 	confFileName := fmt.Sprintf("/tmp/ubiquity-plugin%s.conf", "Sock")
> ```

what is this name Sock?

---

*[ubiquity-client.conf, line 1 at r3](https://reviewable.io:443/reviews/ibm/ubiquity-docker-plugin/53#-Kw5v75jz0tpIhNAfHQq:-Kw5v75kh5xF2YZUloEp:b-izj4ot) ([raw file](https://github.com/ibm/ubiquity-docker-plugin/blob/758ce358f0e128e4edacb736f0feb495eee39775/ubiquity-client.conf#L1)):*
Are all these params still relevant??   I think you put them all inside the json file, don't you?  please clarify.

> = "/tmp"
> backends = ["spectrum-scale"]
> logLevel = "info"         # debug / info / error

---

*[scripts/plugin_v2.sh, line 3 at r3](https://reviewable.io:443/reviews/ibm/ubiquity-docker-plugin/53#-Kw5vUPN7Qmy71LZeZMD:-Kw5vUPOx8gShvVSGoPj:b-apumr9) ([raw file](https://github.com/ibm/ubiquity-docker-plugin/blob/758ce358f0e128e4edacb736f0feb495eee39775/scripts/plugin_v2.sh#L3)):*
> ```Shell
> #!/bin/bash
> 
> set -e
> ```

please rename the file to 
build_docker_plugin.sh

---

*[scripts/plugin_v2.sh, line 5 at r3](https://reviewable.io:443/reviews/ibm/ubiquity-docker-plugin/53#-Kw5vhYd25J9EcAwzpbo:-Kw5vhYd25J9EcAwzpbp:b-m65i7e) ([raw file](https://github.com/ibm/ubiquity-docker-plugin/blob/758ce358f0e128e4edacb736f0feb495eee39775/scripts/plugin_v2.sh#L5)):*
> ```Shell
> set -e
> 
> PLUGIN_NAME="ubiquity/ubiquity-docker-plugin"
> ```

check if this param already exported and if so use it, if not use the default you mentioned here.
and i think the name PLUGIN_NAME is wrong, because it actually the plugin docker repository, right?

---

*[scripts/plugin_v2.sh, line 6 at r3](https://reviewable.io:443/reviews/ibm/ubiquity-docker-plugin/53#-Kw5vpeuUUYbhhK8yovI:-Kw5vpeuUUYbhhK8yovJ:bk9e2g7) ([raw file](https://github.com/ibm/ubiquity-docker-plugin/blob/758ce358f0e128e4edacb736f0feb495eee39775/scripts/plugin_v2.sh#L6)):*
> ```Shell
> 
> PLUGIN_NAME="ubiquity/ubiquity-docker-plugin"
> TAG="v1"
> ```

the same here
please check if TAG export already exist, if so use this TAG if not use v1.0.0 as the default.
also print at the begin of the script the plugin name and the tag

---

*[web_server/handler.go, line 82 at r3](https://reviewable.io:443/reviews/ibm/ubiquity-docker-plugin/53#-Kw5wMokcA_n17PenrLm:-Kw5wJ06vdDpZN9xyCxi:b5jr0lx) ([raw file](https://github.com/ibm/ubiquity-docker-plugin/blob/758ce358f0e128e4edacb736f0feb495eee39775/web_server/handler.go#L82)):*
> ```Go
> 		return
> 	}
> 	removeVolumeRequest.ContainerOrchestrator = "docker"
> ```

please add this "docker" string as const inside the ubiquity repository and use it here instead of duplicate the "docker" here and in the plan you going to use it for validation.

---

*[web_server/handler.go, line 98 at r3](https://reviewable.io:443/reviews/ibm/ubiquity-docker-plugin/53#-Kw5wd4SbasYkjz6h9ws:-Kw5wd4SbasYkjz6h9wt:bec7oho) ([raw file](https://github.com/ibm/ubiquity-docker-plugin/blob/758ce358f0e128e4edacb736f0feb495eee39775/web_server/handler.go#L98)):*
> ```Go
> 	}
> 	attachRequest.Host = c.hostname
> 	attachRequest.ContainerOrchestrator = "docker"
> ```

same here - use const

---

*[web_server/server.go, line 60 at r3](https://reviewable.io:443/reviews/ibm/ubiquity-docker-plugin/53#-Kw5wlUZXpvMN1LOfmxg:-Kw5wlUZXpvMN1LOfmxh:b5m7t5r) ([raw file](https://github.com/ibm/ubiquity-docker-plugin/blob/758ce358f0e128e4edacb736f0feb495eee39775/web_server/server.go#L60)):*
> ```Go
> 	router.HandleFunc("/VolumeDriver.Path", s.handler.Path).Methods("POST")
> 	router.HandleFunc("/VolumeDriver.List", s.handler.List).Methods("POST")
> 	router.HandleFunc("/VolumeDriver.Capabilities", s.handler.GetCapabilities).Methods("POST")
> ```

why capability is post and not GET?

ahhh its because we sending the request - i must say its very confusing.

---

*[web_server/server.go, line 70 at r3](https://reviewable.io:443/reviews/ibm/ubiquity-docker-plugin/53#-Kw5x1sz5xNi0AfMcjTN:-Kw5x1t-fW7iGr_PQqtf:bdzzx0i) ([raw file](https://github.com/ibm/ubiquity-docker-plugin/blob/758ce358f0e128e4edacb736f0feb495eee39775/web_server/server.go#L70)):*
> ```Go
> 	err = s.serveUnixSocket(pluginsPath, router)
> 	if err != nil {
> 		panic("Error starting web server: " + err.Error())
> ```

why  the error message is "error starting web server"??   isn't should be fail to open unix socket?

---

*[web_server/server.go, line 77 at r3](https://reviewable.io:443/reviews/ibm/ubiquity-docker-plugin/53#-Kw5xNFsqNTmSaQ7eL2Q:-Kw5xNFsqNTmSaQ7eL2R:bj224ys) ([raw file](https://github.com/ibm/ubiquity-docker-plugin/blob/758ce358f0e128e4edacb736f0feb495eee39775/web_server/server.go#L77)):*
> ```Go
> 	if _,err := os.Stat(resources.DockerPropagatedMount); err != nil {
> 		if os.IsNotExist(err) {
> 			if err := os.MkdirAll(resources.DockerPropagatedMount, 0755); err != nil {
> ```

why 755?   
do you thing other should be able to cd to this directory?

do we expect this path to be already created?  

---

*[web_server/server.go, line 93 at r3](https://reviewable.io:443/reviews/ibm/ubiquity-docker-plugin/53#-Kw5xwM7iNGxaAEaHBHZ:-Kw5xwM8J1QKGQv0fN2y:b-lwok6d) ([raw file](https://github.com/ibm/ubiquity-docker-plugin/blob/758ce358f0e128e4edacb736f0feb495eee39775/web_server/server.go#L93)):*
> ```Go
> 	if _,err := os.Stat(pluginsPath); err != nil {
> 		if os.IsNotExist(err) {
> 			if err := os.MkdirAll(pluginsPath, 0755); err != nil {
> ```

755 is ok?

---

*[web_server/server.go, line 103 at r3](https://reviewable.io:443/reviews/ibm/ubiquity-docker-plugin/53#-Kw5y3Iy4-yhyp6JzK6H:-Kw5y3Iz6xpHsxtD6WyN:bk5c8cq) ([raw file](https://github.com/ibm/ubiquity-docker-plugin/blob/758ce358f0e128e4edacb736f0feb495eee39775/web_server/server.go#L103)):*
> ```Go
> 	}
> 
> 	ubiquitySocketAddress := path.Join(pluginsPath, fmt.Sprintf("%s.sock", "ubiquity"))
> ```

please put this string :  fmt.Sprintf("%s.sock", "ubiquity") as const

---

*[web_server/server.go, line 105 at r3](https://reviewable.io:443/reviews/ibm/ubiquity-docker-plugin/53#-Kw5yK8mnSQX_wN7I_FJ:-Kw5yK8mnSQX_wN7I_FK:bck80p7) ([raw file](https://github.com/ibm/ubiquity-docker-plugin/blob/758ce358f0e128e4edacb736f0feb495eee39775/web_server/server.go#L105)):*
> ```Go
> 	ubiquitySocketAddress := path.Join(pluginsPath, fmt.Sprintf("%s.sock", "ubiquity"))
> 
> 	err := syscall.Unlink(ubiquitySocketAddress)
> ```

why syscall?


anyway please move this operation inside the ubiquity Executor interface
so you will be able to add unit testing for this part.

did you done unit testing for this code? not sure you can until u put it inside the xecutor

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/ibm/ubiquity-docker-plugin/53)*
<!-- Sent from Reviewable.io -->

shay-berman commented:
Hi @amit
Thanks for the effort of this task.

I conducted code review on this PR, please review and apply them.
In addition - is this the only PR for the new docker plugin v2? 

Thanks

aswarke commented:
Review status: 0 of 13 files reviewed at latest revision, 19 unresolved discussions.

---

*[config.json, line 3 at r3](https://reviewable.io:443/reviews/ibm/ubiquity-docker-plugin/53#-Kw5t-MAls8kxMkLwaxd:-KwpmKhI2pYSFJIdKuOh:bflgzgj) ([raw file](https://github.com/ibm/ubiquity-docker-plugin/blob/758ce358f0e128e4edacb736f0feb495eee39775/config.json#L3)):*
<details><summary><i>Previously, shay-berman wrote…</i></summary><blockquote>

This is problematic
The documentation for GA is going to be in SCBE user guide, so the right location here should be change according for the GA
</blockquote></details>

The documentation for Ubiquity-Docker-plugin should not be SCBE doc. It should be the landing page for docs related to the plugin only. which in our case is the github readme . The documentation link as of now is correct and can be changed in the future when ubiquity has an official docs page

---

*[config.json, line 76 at r3](https://reviewable.io:443/reviews/ibm/ubiquity-docker-plugin/53#-Kw5tYvbvuIxuejqsr8e:-KwpmeZZpdLQxEaYx4V4:b-81ndl3) ([raw file](https://github.com/ibm/ubiquity-docker-plugin/blob/758ce358f0e128e4edacb736f0feb495eee39775/config.json#L76)):*
<details><summary><i>Previously, shay-berman wrote…</i></summary><blockquote>

r u sure about this version 1.0?
</blockquote></details>

Yes, I'm sure about version 1.0, this is version 1.0 of their V2 architecture

---

*[config.json, line 95 at r3](https://reviewable.io:443/reviews/ibm/ubiquity-docker-plugin/53#-Kw5tdDo6vzp0JfNTTzz:-KwpmmHQ44KiXtpdA7fQ:b2garhx) ([raw file](https://github.com/ibm/ubiquity-docker-plugin/blob/758ce358f0e128e4edacb736f0feb495eee39775/config.json#L95)):*
<details><summary><i>Previously, shay-berman wrote…</i></summary><blockquote>

what is this for? 
if its for Spectrum Scale please add docstring here to mention it
</blockquote></details>

This is needed for scale.  There is nothing called docstring . Descriptions etc can be done later during GA of ubiquity-docker-plugin

---

*[config.json, line 113 at r3](https://reviewable.io:443/reviews/ibm/ubiquity-docker-plugin/53#-Kw5tkJY_C5i9opXzdIU:-KwpoBazRjq77ArzNnle:blf8kgh) ([raw file](https://github.com/ibm/ubiquity-docker-plugin/blob/758ce358f0e128e4edacb736f0feb495eee39775/config.json#L113)):*
<details><summary><i>Previously, shay-berman wrote…</i></summary><blockquote>

r u sure we need this bind?  why?  the contain run as root so why we need it?
</blockquote></details>

yes we need to bind /etc/passwd , /etc/group as we have do user, group permissions change for scale backend. 

---

*[Dockerfile, line 10 at r3](https://reviewable.io:443/reviews/ibm/ubiquity-docker-plugin/53#-Kw5tvQzTVJ3BFID4Ni7:-KwpoMg_kww6Qj28XsF8:b-2e8k03) ([raw file](https://github.com/ibm/ubiquity-docker-plugin/blob/758ce358f0e128e4edacb736f0feb495eee39775/Dockerfile#L10)):*
<details><summary><i>Previously, shay-berman wrote…</i></summary><blockquote>

you should mention package=VERSION, so it will take a specific version and not the latest.
so please checkout what are the latest one that it install it with, and just put the vesion inside.  (we want to have full control on the version that comes to GA version)
</blockquote></details>

This can be changed later, it is always advisable to have the latest version of packages to avoid defects and security vulnerabilities

---

*[glide.yaml, line 6 at r3](https://reviewable.io:443/reviews/ibm/ubiquity-docker-plugin/53#-Kw5uOBr9gt-iyGeXwZF:-KwpoghAlg7RU38AgLo7:b-92rdma) ([raw file](https://github.com/ibm/ubiquity-docker-plugin/blob/758ce358f0e128e4edacb736f0feb495eee39775/glide.yaml#L6)):*
<details><summary><i>Previously, shay-berman wrote…</i></summary><blockquote>

change it to dev before the marj
</blockquote></details>

yes, changed to dev 

---

*[main_suite_test.go, line 80 at r3](https://reviewable.io:443/reviews/ibm/ubiquity-docker-plugin/53#-Kw5ulZ6mG0kEAFvqAa_:-KwpokD_K8RzU2LpGbSn:b-c0jsls) ([raw file](https://github.com/ibm/ubiquity-docker-plugin/blob/758ce358f0e128e4edacb736f0feb495eee39775/main_suite_test.go#L80)):*
<details><summary><i>Previously, shay-berman wrote…</i></summary><blockquote>

what is this name Sock?
</blockquote></details>

This is for our integration testing . Integration testing will probably be meaningless going forward as plugin must be tested as v2, so acceptance tests will be a better way to go forward .

---

*[ubiquity-client.conf, line 1 at r3](https://reviewable.io:443/reviews/ibm/ubiquity-docker-plugin/53#-Kw5v75jz0tpIhNAfHQq:-Kwpp-jTZIOHJkjqQOAs:b-s8g93y) ([raw file](https://github.com/ibm/ubiquity-docker-plugin/blob/758ce358f0e128e4edacb736f0feb495eee39775/ubiquity-client.conf#L1)):*
<details><summary><i>Previously, shay-berman wrote…</i></summary><blockquote>

Are all these params still relevant??   I think you put them all inside the json file, don't you?  please clarify.

> = "/tmp"
> backends = ["spectrum-scale"]
> logLevel = "info"         # debug / info / error


</blockquote></details>

Integration tests run differently than acceptance tests. CI pipeline will going forward need to run acceptance tests only . These commits to integration testing are needed but will not be used in CI/ CD pipeline but these changes need to go in.

---

*[scripts/plugin_v2.sh, line 3 at r3](https://reviewable.io:443/reviews/ibm/ubiquity-docker-plugin/53#-Kw5vUPN7Qmy71LZeZMD:-KwppeSzgVaEqe7fXO9G:bb7f2mf) ([raw file](https://github.com/ibm/ubiquity-docker-plugin/blob/758ce358f0e128e4edacb736f0feb495eee39775/scripts/plugin_v2.sh#L3)):*
<details><summary><i>Previously, shay-berman wrote…</i></summary><blockquote>

please rename the file to 
build_docker_plugin.sh
</blockquote></details>

done

---

*[scripts/plugin_v2.sh, line 5 at r3](https://reviewable.io:443/reviews/ibm/ubiquity-docker-plugin/53#-Kw5vhYd25J9EcAwzpbo:-KwpphNu0cQ6MafSafBt:b-kt3ajp) ([raw file](https://github.com/ibm/ubiquity-docker-plugin/blob/758ce358f0e128e4edacb736f0feb495eee39775/scripts/plugin_v2.sh#L5)):*
<details><summary><i>Previously, shay-berman wrote…</i></summary><blockquote>

check if this param already exported and if so use it, if not use the default you mentioned here.
and i think the name PLUGIN_NAME is wrong, because it actually the plugin docker repository, right?
</blockquote></details>

no need to do env checking. This is just a variable  and it is plugin name . Please try it out first to know how it works. 

---

*[scripts/plugin_v2.sh, line 6 at r3](https://reviewable.io:443/reviews/ibm/ubiquity-docker-plugin/53#-Kw5vpeuUUYbhhK8yovI:-KwppwpbxDLVP899kCw6:b-lpbx4j) ([raw file](https://github.com/ibm/ubiquity-docker-plugin/blob/758ce358f0e128e4edacb736f0feb495eee39775/scripts/plugin_v2.sh#L6)):*
<details><summary><i>Previously, shay-berman wrote…</i></summary><blockquote>

the same here
please check if TAG export already exist, if so use this TAG if not use v1.0.0 as the default.
also print at the begin of the script the plugin name and the tag
</blockquote></details>

No need for env checking first . Printing at beginning is already done

---

*[web_server/handler.go, line 82 at r3](https://reviewable.io:443/reviews/ibm/ubiquity-docker-plugin/53#-Kw5wMokcA_n17PenrLm:-KwpqBLSGg5WeKbAltXR:bb7f2mf) ([raw file](https://github.com/ibm/ubiquity-docker-plugin/blob/758ce358f0e128e4edacb736f0feb495eee39775/web_server/handler.go#L82)):*
<details><summary><i>Previously, shay-berman wrote…</i></summary><blockquote>

please add this "docker" string as const inside the ubiquity repository and use it here instead of duplicate the "docker" here and in the plan you going to use it for validation.
</blockquote></details>

done

---

*[web_server/handler.go, line 98 at r3](https://reviewable.io:443/reviews/ibm/ubiquity-docker-plugin/53#-Kw5wd4SbasYkjz6h9ws:-KwpvWUng_wE9jiafjv-:bb7f2mf) ([raw file](https://github.com/ibm/ubiquity-docker-plugin/blob/758ce358f0e128e4edacb736f0feb495eee39775/web_server/handler.go#L98)):*
<details><summary><i>Previously, shay-berman wrote…</i></summary><blockquote>

same here - use const
</blockquote></details>

done

---

*[web_server/server.go, line 60 at r3](https://reviewable.io:443/reviews/ibm/ubiquity-docker-plugin/53#-Kw5wlUZXpvMN1LOfmxg:-KwpvYW1WwqaFwKUTgxP:bz2jnbr) ([raw file](https://github.com/ibm/ubiquity-docker-plugin/blob/758ce358f0e128e4edacb736f0feb495eee39775/web_server/server.go#L60)):*
<details><summary><i>Previously, shay-berman wrote…</i></summary><blockquote>

why capability is post and not GET?

ahhh its because we sending the request - i must say its very confusing.
</blockquote></details>

POST is good for now

---

*[web_server/server.go, line 70 at r3](https://reviewable.io:443/reviews/ibm/ubiquity-docker-plugin/53#-Kw5x1sz5xNi0AfMcjTN:-KwpvbiBEnLyghrT2EAO:bv441a0) ([raw file](https://github.com/ibm/ubiquity-docker-plugin/blob/758ce358f0e128e4edacb736f0feb495eee39775/web_server/server.go#L70)):*
<details><summary><i>Previously, shay-berman wrote…</i></summary><blockquote>

why  the error message is "error starting web server"??   isn't should be fail to open unix socket?
</blockquote></details>

This is correct message. We do capture error message when socket fails and log that.

---

*[web_server/server.go, line 77 at r3](https://reviewable.io:443/reviews/ibm/ubiquity-docker-plugin/53#-Kw5xNFsqNTmSaQ7eL2Q:-KwpqVeaklTJQlGx4K0g:b-r8eequ) ([raw file](https://github.com/ibm/ubiquity-docker-plugin/blob/758ce358f0e128e4edacb736f0feb495eee39775/web_server/server.go#L77)):*
<details><summary><i>Previously, shay-berman wrote…</i></summary><blockquote>

why 755?   
do you thing other should be able to cd to this directory?

do we expect this path to be already created?  
</blockquote></details>

path is created  while creating the plugin, this is a secondary check just in case.

---

*[web_server/server.go, line 93 at r3](https://reviewable.io:443/reviews/ibm/ubiquity-docker-plugin/53#-Kw5xwM7iNGxaAEaHBHZ:-Kwpr1yDNKHOTx_PtasI:b-ed7zgb) ([raw file](https://github.com/ibm/ubiquity-docker-plugin/blob/758ce358f0e128e4edacb736f0feb495eee39775/web_server/server.go#L93)):*
<details><summary><i>Previously, shay-berman wrote…</i></summary><blockquote>

755 is ok?
</blockquote></details>

Yes this is needed and I've checked with Docker folks 

---

*[web_server/server.go, line 103 at r3](https://reviewable.io:443/reviews/ibm/ubiquity-docker-plugin/53#-Kw5y3Iy4-yhyp6JzK6H:-Kwpr98KYg_bDjoR_xJM:bcr86qg) ([raw file](https://github.com/ibm/ubiquity-docker-plugin/blob/758ce358f0e128e4edacb736f0feb495eee39775/web_server/server.go#L103)):*
<details><summary><i>Previously, shay-berman wrote…</i></summary><blockquote>

please put this string :  fmt.Sprintf("%s.sock", "ubiquity") as const
</blockquote></details>

That  is not necessary

---

*[web_server/server.go, line 105 at r3](https://reviewable.io:443/reviews/ibm/ubiquity-docker-plugin/53#-Kw5yK8mnSQX_wN7I_FJ:-KwpvHzdvhXjKFugDwj2:bj71gou) ([raw file](https://github.com/ibm/ubiquity-docker-plugin/blob/758ce358f0e128e4edacb736f0feb495eee39775/web_server/server.go#L105)):*
<details><summary><i>Previously, shay-berman wrote…</i></summary><blockquote>

why syscall?


anyway please move this operation inside the ubiquity Executor interface
so you will be able to add unit testing for this part.

did you done unit testing for this code? not sure you can until u put it inside the xecutor
</blockquote></details>

changed to using executor to use unlink instead of syscall. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/ibm/ubiquity-docker-plugin/53)*
<!-- Sent from Reviewable.io -->
